app/webpacker/controllers/bulk_form_controller.js

```diff
@@ -0,0 +1,90 @@
+import { Controller } from "stimulus";
+
+// Manages "modified" state for a form with multiple records
+export default class BulkFormController extends Controller {
+  static targets = ["actions", "modifiedSummary"];
+  static values = {
+    disableSelector: String,
+  };
+  recordElements = {};
+
+  connect() {
+    this.form = this.element;
+
+    // Start listening for any changes within the form
+    // this.element.addEventListener('change', this.toggleModified.bind(this)); // dunno why this doesn't work
+    for (const element of this.form.elements) {
```

Comment:
Can't you use `this.form.elements.forEach((element) => { ... })`  ?

Reply:
Yes, good pickup, it would be better to use that format.

Reply:
Sadly, it doesn't work. It's a `HTMLFormControlsCollection`, which apparently doesn't implement `forEach`. Casting it to an array works, but it's kinda ugly so I'm not sure it's worth it:

```suggestion
    Array.from(this.form.elements).forEach((element) => { 
```

Reply:
Not really worth it.